iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -203,11 +207,44 @@ public override async Task SendEventAsync(IEnumerable<Message> messages, Cancell
             }
         }
 
-        public override async Task<Message> ReceiveAsync(TimeSpan timeout, CancellationToken cancellationToken)
+        public override async Task<Message> ReceiveAsync(CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
+
             Message message = null;
 
-            cancellationToken.ThrowIfCancellationRequested();
+            this.EnsureValidState();
```

Comment:
Is this common code? If yes, can you avoid code duplication here by extracting a method?

Reply:
Done